lib/buy/onramper/onramper_buy_provider.dart

```diff
@@ -131,13 +167,13 @@ class OnRamperBuyProvider extends BuyProvider {
       required bool isBuyAction,
       required String walletAddress,
       PaymentType? paymentType,
+      String? customPaymentMethodType,
       String? countryCode}) async {
     String? paymentMethod;
 
-    if (paymentType != null && paymentType != PaymentType.all) {
-      paymentMethod = normalizePaymentMethod(paymentType);
-      if (paymentMethod == null) paymentMethod = paymentType.name;
-    }
+    if (paymentType == PaymentType.all && recommendedPaymentType != null) paymentMethod = recommendedPaymentType!;
+    else if (paymentType == PaymentType.unknown) paymentMethod = customPaymentMethodType;
+    else if (paymentType != null) paymentMethod = normalizePaymentMethod(paymentType);
```

Comment:
if payment type is credit card for example. (this is what the user selected)

the first if condition is going to be false.
the second one would probably be true if we already have a custom payment type available, and thus we will set the `paymentMethod` with the custom one, instead of the credit card the user selected

Reply:
In the case of a payment method like credit card, there’s a specific `PaymentType.creditCard,` so this condition (paymentType == PaymentType.unknown) will be false.